File: WebSample/JSViewer_Angular_Core/JSViewer_Angular_Core.Client/src/app/nav-menu/ReportDescriptor.ts

export class ReportDescriptor {
    public name: string;
    public hasSnapshot: boolean;
    public snapshotName: string;
    public useSnapshot: boolean;

  constructor(name: string, hasSnapshot: boolean, snapshotName: string) {
        this.name = name;
        this.hasSnapshot = hasSnapshot;
        this.snapshotName = snapshotName;
        this.useSnapshot = hasSnapshot;
    }
};

File: WebSample/BlazorViewer/BlazorViewerWebAssembly/Data/ReportDescriptor.cs
namespace BlazorViewerWebAssembly.Data
{
	public sealed class ReportDescriptor
	{
		public ReportDescriptor(string name, bool hasSnapshot, string snapshotName)
		{
			Name = name;
			HasSnapshot = hasSnapshot;
			SnapshotName = snapshotName;
		}

		public string Name { get; }
		public bool HasSnapshot { get; }
		public string SnapshotName { get; }
		public bool UseSnapshot { get; set; }
		public string GetReportName() => UseSnapshot ? SnapshotName : Name;
	}
}
